File: refactor_styles.js
const fs = require('fs');
const path = require('path');

function splitFile(filePath, styleFileName) {
  const content = fs.readFileSync(filePath, 'utf-8');
  const styleMatch = content.match(/const styles = StyleSheet\.create\({([\s\S]*)\}\);?/);
  
  if (!styleMatch) {
    console.log(`No styles found in ${filePath}`);
    return;
  }
  
  let beforeStyles = content.slice(0, styleMatch.index).trimRight() + '\n';
  const styleObj = styleMatch[0].replace('const styles =', 'export const styles =');
  const importStatement = `import { styles } from './${styleFileName.replace('.ts', '')}';\n`;
  
  // Enlever StyleSheet des imports du composant si ce n'est plus utilisé ? Optionnel, mais on va l'ajouter au fichier de styles.
  
  // Trouver où injecter l'import
  // le plus simple : après la dernière ligne d'import
  const lastImportIndex = beforeStyles.lastIndexOf('import');
  let nextLineAfterImport = 0;
  if (lastImportIndex !== -1) {
    nextLineAfterImport = beforeStyles.indexOf('\n', lastImportIndex) + 1;
  }
  
  const newComponentContent = beforeStyles.slice(0, nextLineAfterImport) + importStatement + beforeStyles.slice(nextLineAfterImport);
  
  const styleFileContent = `import { StyleSheet } from 'react-native';\n\n${styleObj}\n`;
  
  const dir = path.dirname(filePath);
  fs.writeFileSync(path.join(dir, styleFileName), styleFileContent);
  fs.writeFileSync(filePath, newComponentContent);
  console.log(`Refactored ${filePath}`);
}

splitFile('./Dashboard/app/withdraw.tsx', 'withdraw.styles.ts');
splitFile('./Dashboard/src/components/BalanceCard.tsx', 'BalanceCard.styles.ts');

File: refactor_styles.py
import os
import re

def split_file(file_path, style_file_name):
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()

    match = re.search(r'const styles = StyleSheet\.create\(\{[\s\S]*\}\);?', content)
    if not match:
        print(f"No styles found in {file_path}")
        return

    style_block = match.group(0)
    before_styles = content[:match.start()].rstrip() + '\n'
    
    style_obj = style_block.replace('const styles =', 'export const styles =')
    
    import_statement = f"import {{ styles }} from './{style_file_name.replace('.ts', '')}';\n"
    
    last_import_idx = before_styles.rfind('import ')
    if last_import_idx != -1:
        next_line_after_import = before_styles.find('\n', last_import_idx) + 1
    else:
        next_line_after_import = 0
        
    new_component_content = (
        before_styles[:next_line_after_import] + 
        import_statement + 
        before_styles[next_line_after_import:]
    )
    
    style_file_content = f"import {{ StyleSheet }} from 'react-native';\n\n{style_obj}\n"
    
    dir_name = os.path.dirname(file_path)
    style_path = os.path.join(dir_name, style_file_name)
    
    with open(style_path, 'w', encoding='utf-8') as f:
        f.write(style_file_content)
        
    with open(file_path, 'w', encoding='utf-8') as f:
        f.write(new_component_content)
        
    print(f"Refactored {file_path}")

split_file('./Dashboard/app/withdraw.tsx', 'withdraw.styles.ts')
split_file('./Dashboard/src/components/BalanceCard.tsx', 'BalanceCard.styles.ts')
